renovate.json5

```diff
@@ -0,0 +1,61 @@
+{
+  "extends": ["apollo-open-source", ":semanticCommits"],
+  "enabledManagers": ["npm", "cargo"],
+  "packageRules": [
+    // Bunch up all non-major npm dependencies into a single PR.  In the common case
+    // where the upgrades apply cleanly, this causes less noise and is resolved faster
+    // than starting a bunch of upgrades in parallel for what may turn out to be
+    // a suite of related packages all released at once.
+    //
+    // Since too much in the Rust ecosystem is pre-1.0, we make an exception here.
+    {
+      "matchPackageNames": ["harmonizer"],
+      "groupName": "harmonizer",
+      "groupSlug": "harmonizer"
+    },
+    {
+      "matchCurrentVersion": "< 1.0.0",
+      "separateMinorPatch": true,
+      "matchManagers": ["cargo"],
+      "minor": {
+        "groupName": "cargo pre-1.0 packages",
+        "groupSlug": "cargo-all-pre-1.0"
+      },
+      "patch": {
+        "groupName": "cargo pre-1.0 packages",
+        "groupSlug": "cargo-all-pre-1.0",
+        "automerge": false
+      }
+    },
+    {
+      "matchCurrentVersion": ">= 1.0.0",
+      "matchManagers": ["cargo", "npm"],
+      "matchUpdateTypes": ["minor", "patch", "pin", "digest"],
+      "groupName": "all non-major packages >= 1.0",
+      "groupSlug": "all-non-major-gte-1.0"
+    },
+    // The current Apollo Gatsby theme does not support a version of Gatsby
+    // that supports React 17.
+    {
+      "matchPaths": [
+        "docs/package.json"
+      ],
+      "matchPackageNames": ["react", "react-dom"],
+      "allowedVersions": "16.x",
+    },
+    // The current Apollo Gatsby version is pinned to v2
+    {
+      "matchPaths": [
+        "docs/package.json"
+      ],
+      "matchPackageNames": ["gatsby"],
+      "allowedVersions": "2.x",
+    }
+  ],
+  "pathRules": [
+    {
+      "paths": ["docs/package.json"],
+      "extends": ["apollo-docs"],
+    }
+  ]
```

Comment:
The Federation repo actually has this extension within the `packageRules`, but I actually like/prefer this hierarchy if it works. 

(Though I wish the React hack was in the `apollo-docs` shared config!! That's not a concern of your PR though.)